openedx/core/djangoapps/course_groups/views.py

```diff
@@ -71,6 +89,33 @@ def _get_cohort_representation(cohort, course):
     }
 
 
+@require_http_methods(("GET", "PUT", "POST"))
+@ensure_csrf_cookie
+@expect_json
+@login_required
+def course_cohort_settings_handler(request, course_key_string):
+    """
+    The restful handler for cohort setting requests. Requires JSON.
+    This will raise 404 if user is not staff.
+    GET
+        Returns the JSON representation of cohort settings for the course.
+    PUT or POST
+        Updates the cohort settings for the course. Returns the JSON representation of updated settings.
```

Comment:
I initially used PATCH and it was working great but unable to test PATCH in tests because Django RequestFactory does not support PATCH. So i remove the PATCH.